.github/workflows/enforce-toolchain-synchronization.yml

```diff
@@ -0,0 +1,121 @@
+# Enforce rebasing Pull Requests if Toolchain was modified on target branch
+#
+# If new commits, which modifies toolchain files was pushed to given branch, find all PRs targeting
+# this branch, which also change toolchain files. Then, enforce rebasing them by changing
+# CI/Jenkins/toolchain check to FAILURE.
+# This prevents race condition issue when new toolchain bundle has to be built after PR is merged.
+#
+# Toolchain files:
+# * scripts/requirements-fixed.txt
+# * scripts/tools-versions-darwin.yml
+# * scripts/tools-versions-win10.yml
+# * scripts/tools-versions-linux.yml
+
+name: Enforce rebasing Pull Requests if Toolchain was modified on target branch
+
+on:
+  push:
+    branches:
+      - '**'  # Triggers on pushes to any branch
+
+jobs:
+  check-prs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Define list of files to check
+        id: define_files
+        run: |
+          echo "TOOLCHAIN_FILES=scripts/requirements-fixed.txt,scripts/tools-versions-linux.yml,scripts/tools-versions-darwin.yml,scripts/tools-versions-win10.yml" >> $GITHUB_ENV
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      - name: Get files modified by recent commits
+        id: get_files
+        run: |
+          echo "Modified files in this commit:"
+          git diff --name-only ${{ github.event.before }} ${{ github.sha }} > modified_files.txt
+          cat modified_files.txt
+
+      - name: Check if any watched files are modified
+        id: check_files
+        run: |
+          modified_files=$(cat modified_files.txt)
+          IFS=',' read -r -a watched_files <<< "${{ env.TOOLCHAIN_FILES }}"
+          modified=false
+          for file in "${watched_files[@]}"; do
+            if echo "$modified_files" | grep -q "$file"; then
+              echo "$file was modified."
+              modified=true
+            fi
+          done
+          echo "modified=$modified" >> $GITHUB_ENV
+
+      # App token is required to update Check Status
+      - name: Get jenkins-ncs App token
+        if: env.modified == 'true'
+        uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.JENKINS_NCS_APP_ID }}
```

Comment:
this app has access scope for just writing checks?

Reply:
It's the same App we use to report status from Jenkins